lms/djangoapps/discussion/rest_api/api.py

```diff
@@ -1263,3 +1271,72 @@ def delete_comment(request, comment_id):
         comment_deleted.send(sender=None, user=request.user, post=cc_comment)
     else:
         raise PermissionDenied
+
+
+@function_trace("_get_course_discussion_user_stats")
+def _get_course_discussion_user_stats(
+    course_key: CourseKey,
+    order_by: UserOrdering = UserOrdering.BY_FLAGS,
+) -> List[Dict[str, Union[str, int]]]:
+    """
+    Get course user discussion stats.
+
+    Args:
+        course_key (CourseKey): course key for course to fetch discussion stats for.
+        order_by (UserOrdering): the ordering to use for sorting data
+
+    Returns:
+        Sorted user stat data for the specified course
+    """
+    course_stats = get_course_user_stats(course_key)
+
+    def key_func(user):
+        key = (user["threads"], user["responses"], user["replies"])
+        if order_by == UserOrdering.BY_FLAGS:
+            key = (user["active_flags"], user["inactive_flags"]) + key
+        return key
+
+    return sorted(course_stats.values(), key=key_func, reverse=True)
+
+
+@function_trace("get_course_discussion_user_stats")
+def get_course_discussion_user_stats(
+    request,
+    course_key_str: str,
+    page: int,
+    page_size: int,
+    order_by: UserOrdering = None,
+) -> Dict:
+    """
+    Get paginated course discussion stats for users in the course.
+
+    Args:
+        request (Request): DRF request
+        course_key_str (str): course key string
+        page (int): Page number to fetch
+        page_size (int): Number of items in each page
+        order_by (UserOrdering): The ordering to use for the user stats
+
+    Returns:
+        Paginated data of a user's discussion stats sorted based on the specified ordering.
+
+    """
+    course_key = CourseKey.from_string(course_key_str)
+    if has_discussion_privileges(user=request.user, course_id=course_key):
+        order_by = order_by or UserOrdering.BY_FLAGS
+    else:
+        order_by = order_by or UserOrdering.BY_ACTIVITY
+        if order_by != UserOrdering.BY_ACTIVITY:
+            raise ValidationError({"order_by": ["Invalid value"]})
```

Comment:
@xitij2000 Any reason why value of "order_by" is a _list_  `["Invalid value"]`?

Reply:
It's not necessary for it to be a list, but it can be, to allow for multiple errors on a single field.
i.e. a field can fail multiple validation, a password can be too short and also not have a numeric character. 

I don't anticipate multiple errors here so will just make it a string.